packages/web-components/src/wallet-connection/makeAgoricSigner.js

```diff
@@ -0,0 +1,123 @@
+// @ts-check
+import { fromBech32, toBase64 } from '@cosmjs/encoding';
+import { assertIsDeliverTxSuccess } from '@cosmjs/stargate';
+import { stableCurrency } from './chainInfo.js';
+import { AgoricMsgs } from './signerOptions.js';
+
+/** @typedef {import("@cosmjs/proto-signing").EncodeObject} EncodeObject */
+/** @typedef {import("@cosmjs/stargate").AminoConverters} AminoConverters */
+/** @typedef {import("@cosmjs/stargate").StdFee} StdFee */
+/** @typedef {import('@keplr-wallet/types').ChainInfo} ChainInfo */
+/** @typedef {import('@keplr-wallet/types').Keplr} Keplr */
+/** @typedef {import('@cosmjs/stargate').SigningStargateClient} SigningStargateClient */
+
+/**
+ * @param {string} address
+ * @returns {Uint8Array}
+ */
+const toAccAddress = address => {
+  return fromBech32(address).data;
+};
+
+// XXX domain of @agoric/cosmic-proto
+/**
+ * non-exhaustive list of powerFlags
+ *
+ * See also MsgProvision in golang/cosmos/proto/agoric/swingset/msgs.proto
+ */
+const PowerFlags = {
+  SMART_WALLET: 'SMART_WALLET',
+};
+
+/** @typedef {{owner: string, spendAction: string}} WalletSpendAction */
+
+/**
+ * @returns {StdFee}
+ */
+const zeroFee = () => {
+  const { coinMinimalDenom: denom } = stableCurrency;
+  const fee = {
+    amount: [{ amount: '0', denom }],
+    gas: '300000', // TODO: estimate gas?
+  };
+  return fee;
+};
+
+/**
+ * Use a signing client to
+ * @param {SigningStargateClient} signingClient
+ * @param {string} address
+ * Ref: https://docs.keplr.app/api/
+ */
+export const makeAgoricSigner = (signingClient, address) => {
+  const fee = zeroFee();
+
+  return harden({
+    /**
+     * Sign and broadcast Provision for a new smart wallet
+     *
+     * @throws if account does not exist on chain, user cancels,
+     *         RPC connection fails, RPC service fails to broadcast (
+     *         for example, if signature verification fails)
+     */
+    provisionSmartWallet: async () => {
+      const { accountNumber, sequence } = await signingClient.getSequence(
+        address,
+      );
+      console.log({ accountNumber, sequence });
+
+      const b64address = toBase64(toAccAddress(address));
+
+      const act1 = {
+        typeUrl: AgoricMsgs.MsgProvision.typeUrl,
+        value: {
+          address: b64address,
+          nickname: 'my wallet',
+          powerFlags: [PowerFlags.SMART_WALLET],
+          submitter: b64address,
+        },
+      };
```

Comment:
Worth exporting `makeProvisionSmartWalletMsg(address)` along with the types and converters?

Reply:
I'm not sure why someone would need this exported if they can just use the wallet connection object. I could be convinced otherwise, but wanted to keep these changes as minimal as possible for the goal of the wallet connection thing. Ditto for `makeWalletSpendMsg`

Reply:
Thinking here was to have exports for folks in this camp, but don't feel strongly about it. Maybe these folks can use `@agoric/cosmic-proto` which is being updated to use telescope.
>  My intention is if someone's using a signer for non-agoric stuff also, they should provide their own signing client with the help of agoricRegistryTypes/agoricConverters.